operator/internal/handlers/internal/storage/secrets.go

```diff
@@ -105,12 +104,47 @@ func extractS3ConfigSecret(s *corev1.Secret) (*storage.S3StorageConfig, error) {
 	// Extract and validate optional fields
 	region := s.Data["region"]
 
+	sseCfg, err := extractS3SSEConfig(s.Data)
+	if err != nil {
+		return nil, err
+	}
+
 	return &storage.S3StorageConfig{
 		Endpoint:        string(endpoint),
 		Buckets:         string(buckets),
 		AccessKeyID:     string(id),
 		AccessKeySecret: string(secret),
 		Region:          string(region),
+		SSE:             sseCfg,
+	}, nil
+}
+
+func extractS3SSEConfig(d map[string][]byte) (storage.S3SSEConfig, error) {
+	var (
+		sseType                    storage.S3SSEType
+		kmsKeyId, kmsEncryptionCtx string
+	)
+
+	switch sseType = storage.S3SSEType(d["sse_type"]); sseType {
+	case storage.SSEKMSType:
+		kmsEncryptionCtx = string(d["sse_kms_encryption_context"])
+		kmsKeyId = string(d["sse_kms_key_id"])
```

Comment:
Do we want to do any validation on the key-id? It looks like it can either be an arn or just the UUID of the key (tested it with both).

Reply:
I think this belongs into the AWS client go implementation to be tested.